clang/lib/AST/Decl.cpp

```diff
@@ -4505,6 +4505,14 @@ bool FieldDecl::isZeroSize(const ASTContext &Ctx) const {
   if (!CXXRD->isEmpty())
     return false;
 
+  // MS ABI: nonzero if class type with class type fields
```

Comment:
Does any differentiation need to be made between the spellings on 4489?  Additionally, can you elaborate on the comment here?  How can we look at the fields of `CXXRD` if it is empty?

Reply:
As far as I can tell the cases above still apply to MSVC so I think on 4489 it's correct to account for both spellings. Not sure what you mean about the comment? The existing comment above the `isEmpty` line describes what that checks for